Fix groups in status updates

AnandajithS opened the pull request:
Closes issue #70 

This PR fixes the issue of groups shown in status updates and instead displays members by their tracks.

AnandajithS committed:
Fixed status update

chimnayajith commented on src/graphql/models.rs:
```diff
@@ -42,10 +42,9 @@ pub struct Member {
     #[serde(rename = "discordId")]
     pub discord_id: String,
-    #[serde(rename = "groupId")]
-    pub group_id: i32,
```
The `group_id` field is still necessary because first-year students are assigned to groups before being placed into tracks. Even though this PR doesn’t currently handle that distinction, we shouldn't remove the field.

AnandajithS replied:
I've added back the groupId and I've also modified  the query to fetch members to return their respective tracks

chimnayajith reviewed (commented):
Thanks for the PR! Overall, it looks good with just a small suggestion.

I noticed that queries.rs hasn’t been changed in this PR, but I think it needs some updates related to this change -specifically, the query to fetch members is not fetching the track from root.

chimnayajith commented:
Thanks for the update! Just one quick request: could you squash the commits into a single one to keep the commit history clean?

Other than that, LGTM!

hrideshmg commented:
I wonder what will happen when a member does not have a track assigned to them, like for eg - first years. Wouldn't it just print "Track - "? We should probably have some logic to support this edge case.

AnandajithS committed:
Handle null values for track

AnandajithS commented:
> I wonder what will happen when a member does not have a track assigned to them, like for eg - first years. Wouldn't it just print "Track - "? We should probably have some logic to support this edge case.

I've made the required changes
 - Updated the type of `track` to `Option<String>` to handle null values
 - Added a `match` statement in `format_defaulters` to properly handle `track`

hrideshmg commented:
Looks good so far, guess we'll see if it works tonight :)
